urlc175f05cb8_mchllbrt_PermCode_tgz-pJ8-permlib_examples_UniversalSeparableJ8/src/permlab/ui/ClassEnumerationFrame.java

```diff
@@ -0,0 +1,371 @@
+package permlab.ui;
+
+import permlib.Permutation;
+import permlib.classes.InvolutionPermClass;
+import permlib.classes.PermutationClass;
+import permlib.classes.SimplePermClass;
+import java.awt.*;
+import java.util.*;
+import javax.swing.*;
+import permlib.classes.PermClassInterface;
+import permlib.property.Simple;
+import permlib.utilities.IOUtilities;
+
+/**
+ * Frame containing options to animate or enumerate a class of permutation with
+ * specified properties.
+ * 
+ * @author M Albert
+ */
+public class ClassEnumerationFrame extends javax.swing.JFrame {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This class references 48 other classes. By comparison, 98% of the classes in the CodeGuru reference dataset reference fewer. This indicates that this class is highly coupled with other classes. A class that is highly coupled with other classes is difficult to understand and its behavior might change unexpectedly when one of its referenced classes is updated. High coupling could also increase the integration test complexity, maintenance cost and technical debt. We recommend that you simplify this class or break it into multiple classes.